utils/litemap/src/serde.rs

```diff
@@ -9,10 +9,170 @@ use core::marker::PhantomData;
 use serde::de::{MapAccess, SeqAccess, Visitor};
 use serde::{Deserialize, Deserializer};
 
-#[cfg(feature = "serde_serialize")]
+#[cfg(feature = "serde")]
 use serde::{ser::SerializeMap, Serialize, Serializer};
 
-#[cfg(feature = "serde_serialize")]
+#[cfg(feature = "serde")]
+fn is_valid_json_key<T: Serialize + ?Sized>(k: &T) -> bool {
```

Comment:
I would not be opposed to sticking this utility function in `icu_provider::serde`, where we have a few other serde util functions like `icu_provider::serde::borrow_de_utils`

https://unicode-org.github.io/icu4x-docs/doc/icu_provider/serde/index.html

Reply:
Actually that doesn't really help us here, because these are two of the crates that don't depend on icu_provider. Hmm.